src/DAO/UploadDAO.php

```diff
@@ -0,0 +1,220 @@
+<?php
+/**
+ * This File is part of JTL-Software
+ *
+ * User: pkanngiesser
+ * Date: 24.08.18
+ */
+
+namespace JTL\Onetimelink\DAO;
+
+
+use RedBeanPHP\OODBBean;
+use RedBeanPHP\R;
+
+class UploadDAO
+{
+    /**
+     * @var string
+     */
+    private $token;
+
+    /**
+     * @var int
+     */
+    private $receivedBytes;
+
+    /**
+     * @var int
+     */
+    private $maxUploadSize;
+
+    /**
+     * @var bool
+     */
+    private $done;
+
+    /**
+     * @var string
+     */
+    private $identifier;
+
+    /**
+     * @var string
+     */
+    private $created;
+
+    function __construct($token, $receivedBytes = 0, $maxUploadSize = 0, bool $done = false, string $identifier = null, string $created = null)
+    {
+        $this->token = $token;
+        $this->receivedBytes = $receivedBytes;
+        $this->maxUploadSize = $maxUploadSize;
```

Comment:
The upload is stored in the DB because it can be individual for every upload token. You can set a default limit in the config, but it can be extended for an individual user. The upload limit for guest links can also be different. This way the UploadCommand doesn't need to know about all this stuff. It just needs to compare the received_bytes and the max_upload_size. Also I think it would cause more problems if you would be able to change the upload limit while someone is uploading. All running uploads would be aborted if they reach the new limit because the UI told the user he is able to upload this much. If it is necessary to limit the size of running uploads we can maybe implement this as command which updates the limits in the database.